libazureinit/src/provision/ssh.rs

```diff
@@ -67,20 +68,61 @@ pub(crate) fn get_authorized_keys_path_from_sshd() -> io::Result<Option<PathBuf>
     Ok(None)
 }
 
+pub(crate) fn update_sshd_config() -> io::Result<()> {
+    let sshd_config_path = "/etc/ssh/sshd_config";
+    let temp_sshd_config_path = "/etc/ssh/sshd_config.tmp";
```

Comment:
Rather than hard-coding this, let's provide the `sshd_config_path` as a parameter. This makes it easy to write a test for it, and it also means we can potentially configure this (via the work in #137)

Also, we could probably also create the temporary file in a directory made by [tempfile](https://docs.rs/tempfile/latest/tempfile/struct.TempDir.html). This way if something goes wrong, we don't leave a file in /etc/ssh/.

Reply:
We should limit support for this to a drop-in config:
/etc/ssh/sshd_config.d/50-azure-init.conf

We shouldn't be tampering the main config if we can avoid it.

Reply:
Using the suggested /etc/ssh/sshd_config.d/50-azure-init.conf.